lib/lz4hc.c

```diff
@@ -1287,8 +1287,13 @@ static int LZ4HC_compress_optimal ( LZ4HC_CCtx_internal* ctx,
                                     const dictCtx_directive dict,
                                     const HCfavor_e favorDecSpeed)
 {
+    int retval = 0;
 #define TRAILING_LITERALS 3
+#ifdef LZ4HC_HEAPMODE
+    LZ4HC_optimal_t* opt = (LZ4HC_optimal_t*)malloc(sizeof(LZ4HC_optimal_t) * (LZ4_OPT_NUM + TRAILING_LITERALS));
```

Comment:
minor : we like to `const` everything that can be.
In this case, it means `LZ4HC_optimal_t* const opt =`,
meaning the content of the pointer is mutable,
while the pointer itself is fixed for the lifetime of this variable,?
which is a useful signal from a reviewer/reader perspective.

Reply:
Important : an issue here is that `malloc()` may fail,
and as a consequence, we may end up with `opt == NULL`.

You'll have to test that outcome, and exit properly when it happens, otherwise it becomes a security liability.

Due to the "declaration before statement" rule in C90, you'll have to test after all declarations are completed.